Storage/Processor/Processor.php

```diff
@@ -41,31 +44,17 @@ public function getResponse(Request $request, string $hash, string $configHash,
 
         $handler = $this->getResource($config, $filename, $request->headers->getCacheControlDirective('no-cache') === true);
 
-        if ($handler instanceof StreamInterface) {
-            $callback = function () use ($handler) {
-                if ($handler->isSeekable()) {
-                    $handler->rewind();
-                }
-                if (!$handler->isReadable()) {
-                    echo $handler;
-                    return;
-                }
-                while (!$handler->eof()) {
-                    echo $handler->read(8192);
-                }
-            };
-        } else {
-            $callback = function () use ($handler) {
-                while (!feof($handler)) {
-                    print fread($handler, 8192);
-                }
-            };
+        if (! $handler instanceof StreamInterface) {
+            $handler = new Stream($handler);
```

Comment:
If I had to define the ems StorageInterface now I would define functions returning StreamInterface. The fact that every binary can be passed via a Stream is not connected to the fact that this class is an asset processor.
In which cases we do not want a Stream to handle our stuff? A stream just means that we start to send bytes as soon as we can.

Reply:
Ok, it is just that this check (line 47-48) is forcing it to be a Stream. So you always have a Stream in this case

Reply:
Yes indeed. Currently storage services are returning or an ressource or a StreamInterface object. Do you want that I add a deprecated mlessage in case of ressource?